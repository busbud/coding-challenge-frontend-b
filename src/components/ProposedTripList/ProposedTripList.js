// @flow
import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { Typography } from '@material-ui/core';
import { ProposedTrip } from '../ProposedTrip';

import type { ProposedTrip as ProposedTripType } from '../../types';

type Classes = {|
  proposedTrips: string,
|};

type Props = {|
  classes: Classes,
  proposedTrips: Array<ProposedTripType>,
|};

const styles = {
  proposedTrips: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
};

const UnStyledProposedTripList = (props: Props) => {
  const { proposedTrips, classes } = props;

  const renderProposedTrip = () => proposedTrips.map(proposedtrip => (
    <ProposedTrip
      key={`${proposedtrip.arrivalTime}/${proposedtrip.departureTime}`}
      {...proposedtrip}
    />
  ));

  return (
    <div className={classes.proposedTrips}>
      {' '}
      {proposedTrips.length === 0 ? (
        <Typography variant="subheading">No result found </Typography>
      ) : (
        renderProposedTrip()
      )}
    </div>
  );
};

export const ProposedTripList = withStyles(styles)(UnStyledProposedTripList);
