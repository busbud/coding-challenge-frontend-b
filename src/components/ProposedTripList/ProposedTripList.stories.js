// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import { ProposedTripList } from './ProposedTripList';
import { proposedTripList } from './fixtures/proposedTripList';

storiesOf('ProposedTripList', module)
  .add('with Results', () => <ProposedTripList proposedTrips={proposedTripList} />)
  .add('without Result', () => <ProposedTripList proposedTrips={[]} />);
