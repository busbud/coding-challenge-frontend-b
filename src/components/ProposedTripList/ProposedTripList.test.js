import React from 'react';

import renderer from 'react-test-renderer';
import { proposedTripList } from './fixtures/proposedTripList';

import { ProposedTripList } from './ProposedTripList';

describe('LocationSelector', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ProposedTripList proposedTrip={proposedTripList} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
