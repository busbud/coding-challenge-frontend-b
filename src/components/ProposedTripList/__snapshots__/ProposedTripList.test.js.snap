// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposedTripList renders correctly 1`] = `
<div
  className="UnStyledProposedTripList-proposedTrips-1"
>
   
  <div
    className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiCard-root-8 UnStyledProposedTrip-proposedTrip-2"
  >
    <div
      className="MuiCardContent-root-36"
    >
      <div
        className="MuiCardMedia-root-37 UnStyledProposedTrip-media-3"
        style={
          Object {
            "backgroundImage": "url(\\"https://via.placeholder.com/150\\")",
          }
        }
        title="Paella dish"
      />
      <div
        className="UnStyledProposedTrip-tripInformations-4"
      >
        <div
          className="UnStyledProposedTrip-timeInformations-5"
        >
          <p
            className="MuiTypography-root-39 MuiTypography-body1-48"
          >
            10:26 AM
          </p>
          -&gt;
          <p
            className="MuiTypography-root-39 MuiTypography-body1-48"
          >
            10:26 AM
          </p>
        </div>
        <div>
          <h6
            className="MuiTypography-root-39 MuiTypography-subtitle1-57"
          >
            $960.00
          </h6>
          <p
            className="MuiTypography-root-39 MuiTypography-body1-48 MuiTypography-colorTextSecondary-72 UnStyledProposedTrip-travellerCount-6"
          >
            For 9 traveller(s)
          </p>
        </div>
      </div>
      <div
        className="UnStyledProposedTrip-tripComplementatryInformations-7"
      >
        <p
          className="MuiTypography-root-39 MuiTypography-body1-48"
        >
          <span>
            From: 
          </span>
          In front of the bakery
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiCard-root-8 UnStyledProposedTrip-proposedTrip-2"
  >
    <div
      className="MuiCardContent-root-36"
    >
      <div
        className="MuiCardMedia-root-37 UnStyledProposedTrip-media-3"
        style={
          Object {
            "backgroundImage": "url(\\"https://via.placeholder.com/150\\")",
          }
        }
        title="Paella dish"
      />
      <div
        className="UnStyledProposedTrip-tripInformations-4"
      >
        <div
          className="UnStyledProposedTrip-timeInformations-5"
        >
          <p
            className="MuiTypography-root-39 MuiTypography-body1-48"
          >
            10:26 AM
          </p>
          -&gt;
          <p
            className="MuiTypography-root-39 MuiTypography-body1-48"
          >
            10:26 AM
          </p>
        </div>
        <div>
          <h6
            className="MuiTypography-root-39 MuiTypography-subtitle1-57"
          >
            $960.00
          </h6>
          <p
            className="MuiTypography-root-39 MuiTypography-body1-48 MuiTypography-colorTextSecondary-72 UnStyledProposedTrip-travellerCount-6"
          >
            For 8 traveller(s)
          </p>
        </div>
      </div>
      <div
        className="UnStyledProposedTrip-tripComplementatryInformations-7"
      >
        <p
          className="MuiTypography-root-39 MuiTypography-body1-48"
        >
          <span>
            From: 
          </span>
          in the middle of nowhere, Center Park, Iowa
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiCard-root-8 UnStyledProposedTrip-proposedTrip-2"
  >
    <div
      className="MuiCardContent-root-36"
    >
      <div
        className="MuiCardMedia-root-37 UnStyledProposedTrip-media-3"
        style={
          Object {
            "backgroundImage": "url(\\"https://via.placeholder.com/150\\")",
          }
        }
        title="Paella dish"
      />
      <div
        className="UnStyledProposedTrip-tripInformations-4"
      >
        <div
          className="UnStyledProposedTrip-timeInformations-5"
        >
          <p
            className="MuiTypography-root-39 MuiTypography-body1-48"
          >
            10:26 AM
          </p>
          -&gt;
          <p
            className="MuiTypography-root-39 MuiTypography-body1-48"
          >
            10:26 AM
          </p>
        </div>
        <div>
          <h6
            className="MuiTypography-root-39 MuiTypography-subtitle1-57"
          >
            $960.00
          </h6>
          <p
            className="MuiTypography-root-39 MuiTypography-body1-48 MuiTypography-colorTextSecondary-72 UnStyledProposedTrip-travellerCount-6"
          >
            For 18 traveller(s)
          </p>
        </div>
      </div>
      <div
        className="UnStyledProposedTrip-tripComplementatryInformations-7"
      >
        <p
          className="MuiTypography-root-39 MuiTypography-body1-48"
        >
          <span>
            From: 
          </span>
          Between here and there 
        </p>
      </div>
    </div>
  </div>
</div>
`;
